css/css-display/display-contents-accessibility.html

```diff
@@ -0,0 +1,321 @@
+
+    <!-- <img> renders "display:contents" the same as "display:none", see: https://www.w3.org/TR/css-display-3/#unbox-html-->
+    <label for="label-for-button">label</label>
+    <button id="label-for-button" style="display: contents;" data-expectedlabel="label" data-testname="button with display:contents is labelled via label for/id association" class="ex-label"></button>
+    <!-- SVG and display: contents behavior: https://github.com/w3c/svgwg/issues/305-->
+    <svg viewbox="0 0 100 50">
+        <g style="display:contents;" fill="silver" stroke="blue" data-expectedlabel="group label" data-testname="g element with display:contents, as child of svg, is labelled via title element" class="ex-label">
+          <title>group label</title>
+          <circle cx="20" cy="20" r="10" />
+          <circle cx="20" cy="20" r="10" />
+        </g>
+      </svg>
+
+    <a href="#" style="display: contents;" data-expectedlabel="label" data-testname="link with display:contents is labelled via name from contents" class="ex-label">label</a>
+    <button style="display: contents;" data-expectedlabel="label" data-testname="button with display:contents is labelled via name from contents" class="ex-label">label</button>
+    <h3 style="display: contents;" data-expectedlabel="label" data-testname="h3 with display:contents is labelled via name from contents" class="ex-label">label</h3>
+
+    <h2>Accessibility roles</h2>
+    <!-- Buttons -->
+    <button style="display: contents;" data-expectedrole="button" data-testname="button with display:contents has button role" class="ex">x</button>
+    <button role="button" style="display: contents;" data-expectedrole="button" data-testname="button with explicit button role and display:contents has button role" class="ex">x</button>
+    <div role="button" style="display: contents;" data-expectedrole="button" data-testname="div with button role and display:contents has button role" class="ex">x</div>
+    <div role="button" tabindex="0" style="display: contents;" data-expectedrole="button" data-testname="div with button role, tabindex=0 and display:contents has button role" class="ex">x</div>
+
+    <!-- Display: Flex -->
+    <table style="display: flex; flex-flow: column nowrap;">
+        <thead>
+          <tr style="display: contents;" data-expectedrole="row" data-testname="tr with display:contents, within table with display:flex, has row role" class="ex">
+            <th data-expectedrole="rowheader" data-testname="th as child of tr with display:contents, within table with display:flex, has rowheader role" class="ex">x</th>
+            <th>x</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr style="display: contents;">
+            <td data-expectedrole="cell" data-testname="td as child of tr with display:contents, within table with display:flex, has cell role" class="ex">x</td>
```

Comment:
I think Chromium's expectation (`!EQ("cell", "gridcell")`) may be reasonable since the `<table>` does not not explicitly define whether it's an interactive `grid` or a static `table`.  I'd suggest making this `<th role="rowheader">` and changing the expectation to that, leaving the first row's first header a columnheader test. 

I'm not sure if you could get all engines to match "cell" or "gridcell" but I would try to explicitly add a role to the `<table>` and see how that turns out. 

As is, these are 3 different results that may all be reasonable, given the current ambiguity of the test case.
- Chromium thinks this is a grid, so `gridcell` is okay.
- Gecko thinks this is a table, so `cell` is okay.
- WebKit might think this is layout table, so its no role might also be okay.

An explicit `role="grid"` or `role="table"` on the `<table>` is the first resolution I'd try.

Reply:
> I think Chromium's expectation (!EQ("cell", "gridcell")) may be reasonable

To clarify, I am not saying Chromium's is correct and the others are wrong... I'm saying the example leaves room for interpretation so that both expectations may be correct.

Reply:
The `<table>` has no `role` attribute, so:

https://w3c.github.io/html-aam/#el-table says the role for `<table>` should be `table`
https://w3c.github.io/html-aam/#el-td says the role for `<td>` should be `cell`

Reply:
@cookiecrook @zcorpan I see; I've modified these `display: flex` tests to test a `<table>` without explicit role, `<table role="table">` and `<table role="grid">`:  https://github.com/web-platform-tests/wpt/pull/43740/commits/990743931eac4319211bab5404748351c6b40f54.